RTech.CoffeeMachine.Api/BrewCoffee/Controllers/CoffeeMachineController.cs

```diff
@@ -12,54 +12,48 @@ public class CoffeeMachineController : ControllerBase
     private readonly ILogger<CoffeeMachineController> _logger;
     private readonly IDateTimeProvider _dateTimeProvider;
     private readonly IBrewStatusProvider _brewStatusProvider;
-    // Interviewer note: MemoryCache from Microsoft.Extensions.Caching.Abstractions is thread safe
-    private readonly IMemoryCache _cache;
-    public static readonly string CACHE_KEY_NUM_REQUESTS = $"{nameof(CoffeeMachineController)}:numRequests";
 
     public CoffeeMachineController(
         ILogger<CoffeeMachineController> logger,
         IDateTimeProvider dateTimeProvider,
-        IBrewStatusProvider brewStatusProvider,
-        IMemoryCache cache
+        IBrewStatusProvider brewStatusProvider
     )
     {
         _logger = logger;
         _dateTimeProvider = dateTimeProvider;
         _brewStatusProvider = brewStatusProvider;
-        _cache = cache;
-
-        if (!_cache.TryGetValue<long>(CACHE_KEY_NUM_REQUESTS, out var _))
-        {
-            _cache.Set<long>(CACHE_KEY_NUM_REQUESTS, 0);
-        }
     }
 
+    [ServiceFilter(typeof(TeapotFilter))]
+    [ServiceFilter(typeof(BrewStatusUnavailableFilter))]
```

Comment:
I've used `ServiceFilter` instead of `TypeFilter` because I wanted them in the DI container as singletons (rather than transient)